llvm/include/llvm/Target/AIETarget.td

```diff
@@ -20,15 +20,18 @@ class PreSchedInstExpansion<Instruction Target, Instruction Barrier> {
 }
 
 // Scheduling information for instructions that touch memory.
-class MemoryCycles<int First, int Last = -1> {
+class MemoryCycles<list<int> MemCycles> {
+  // Cycles in which memory is accessed 
+  list<int> MemCyclesList = MemCycles;
```

Comment:
Changes done as part of this commit captures all the cycles in which memory is accessed as a list. The idea of firstMemCycle & LastMemCycle is still kept intact.

We will be using this list of cycles to reserve bank in those cycles.  

This approach is future proof in event we have instruction accessing memory in more than two cycles.

Reply:
I think that you can complement the comments with this rationale.